src/quo2/components/community/token_gating.cljs

```diff
@@ -1,295 +1,45 @@
-           [channel-avatar/channel-avatar
-            {:big?                   true
-             :locked?                locked?
-             :emoji                  emoji
-             :emoji-background-color emoji-background-color}])]
-
-        [rn/view {:style (get styles :header-title-container)}
-         [text/text
-          {:weight          :semi-bold
-           :number-of-lines 1
-           :size            :heading-1
-           :style           (get styles :header-text)} (if (= type :community) name (str "# " name))]
-
-         (when (= type :community)
-           [icon/icon
-            (if locked?
-              :main-icons2/locked
-              :main-icons2/unlocked)
-            {:container-style (get styles :header-title-lock)
-             :color           (colors/theme-colors
-                               colors/neutral-50
-                               colors/neutral-40)}])]
-
-        [button/button
-         {:type  :outline
-          :size  32
-          :icon  true
-          :style (get styles :header-info-button)} :main-icons2/info]]
-
-       (map-indexed (fn [gate-index gate]
-                      ^{:key gate-index}
-                      [:<>
-                       (when-not (= gate-index 0)
-                         [rn/view
-                          {:style (merge
-                                   (get styles :divider)
-                                   {:background-color (colors/theme-colors
-                                                       colors/neutral-10
-                                                       colors/neutral-80)})}])
-                       [token-requirement-list
-                        {:gate                        gate
-                         :token-requirements-changed? token-requirements-changed?
-                         :required-tokens-lost?       required-tokens-lost?} community-color]])
-                    gates)
-
-       (when (= type :channel)
-         (if membership-request-denied?
-           [information-box/information-box
-            {:type           :error
-             :icon           :main-icons2/untrustworthy
-             :no-icon-color? true
-             :style          (get styles :membership-request-denied)}
-            (i18n/label
-             :t/membership-request-denied)]
-
-           [:<>
-            [button/button
-             {:type                 :community
-              :community-color      community-color
-              :community-text-color community-text-color
-              :style                (get styles :enter-button)
-              :disabled             locked?
-              :on-press             on-enter-channel}
-             (str "# "
-                  (i18n/label
-                   :t/enter-channel))]
+  [{:keys [tokens padding?]}]
+  [:<>
+   (if (> (count tokens) 1)
+     (doall
+      (map-indexed
+       (fn [token-requirement-index tokens]
+         ^{:key (str "token-group" token-requirement-index)}
```

Comment:
Same here, `(str "token-group" token-requirement-index` I believe is the same as ^{:key token-requirement-index}